components/Masonry.js

```diff
@@ -97,7 +97,7 @@ export default class Masonry extends Component {
 
   render() {
     return (
-  	<View onLayout={(event) => this._setParentDimensions(event)}>
+  	<View style={{flex: 1}} onLayout={(event) => this._setParentDimensions(event)}>
  	    <ListView
```

Comment:
@brh55 This is probably a huge n00b question but why not use a `FlatList` here too? The rendering of things and `flexbox` is still a bit of black magic so I'm still having a hard time conceptualizing Masonry's anatomy. I was checking out Spencer Carli's [blog post](https://medium.com/react-native-development/how-to-use-the-flatlist-component-react-native-basics-92c482816fe6), [(video here)](https://www.youtube.com/watch?v=r-ENJLGrd3s&t=228s) and `FlatList` looks like it takes care of lazy loading, header-footers, pull to refresh. Basically every thing I'm looking for, exception masonry bricks style.

Could you please help me visualize the anatomy of Masonry and other challenges to make this happen?

Reply:
We could use a `FlatList` here, I wasn't aware that the depreciated ListView. We just need to manage state changes within `Masonry` to prompt reloads.  But let me answer your previous questions:

Hopefully this addresses your second question about flexbox / flatlist. Typically, using FlatList + Columns, it *theoretically* renders a row using something similar to a layout like such: 
```
|       FlatList          |
---------------------------
| |      ROW 1 (FlexBox)  | | 
| |   col     |     col   | |
| | --------------------- | |
| |      ROW 2 (FlexBox)  | | 
| |   col     |     col   | |
| | --------------------- | |
```

This makes sense, because FlatList is made to be ideal for rendering a simple list. The issue is when you account for images of different heights, you will have spacing issues as the height of the row will always be the height of the tallest column -- not a desired "masonry" effect. However, if you are attempting to render images of the same height, then `FlatList` + column prop is the way to go.

Now to overcome this issue, we essentially use nested FlatList to act as columns. This is how Masonry's anatomy works *(as of this branch, previously it was just multiple `View`'s)*.

```
|            Masonry (FlatList)        |
----------------------------------------
| Column(FlatList) |  Column(FlatList) |
```

Hopefully that makes some sense, let me know where I can refine on this.

Reply:
One thing I've been wanting to do is to allow anyone to render their own components instead of a "header", "image", "footer".  In most cases this makes it easier to work with for beginners, but this doesn't need to be done like this. 

`Masonry` currently does the calculations of new image sizes based on column widths, but theoretically this could just be passed over to each `renderMethod` and have users calculate that themselves so they can render practically anything. This would be the "masonry-core" component, then we would have something like this currently to be user-friendly.

Reply:
Thanks for the details! Take a deep breathe haha! That was great. I like the idea of rendering passed components too but maybe it would be good to nail down what you currently have. Then look at tackling that idea (I am a beginner in RN :smile:). 


Do you think a refactor is needed to use a `FlatList` at the `Masonry.js` level and ditch the deprecated `ListView`?

Reply:
@kmcgill88 Oops, I thought this is merged.
@brh55 I'd say to just get this working with ```FlatList``` and merge it with master. It would solve performance and infinite scroll issues which have been listed in the issues section. Custom rendering can be on the next version.